compiler/resolution/src/org/jetbrains/kotlin/resolve/calls/components/ResolutionParts.kt

```diff
@@ -635,7 +637,8 @@ internal object CheckReceivers : ResolutionPart() {
                 )
             }
             else -> {
-                searchForAdditionalReceivers()
+                val additionalReceiverArguments = searchForAdditionalReceivers() ?: return
+                resolvedCall.additionalReceiversArguments.addAll(additionalReceiverArguments)
```

Comment:
I'd made `additionalReceiversArguments` to be a read-only list, but add a setter that replaces the whole list